### ISSUE TITLE
Nether Issue

### ISSUE THREAD

**Issue by DeathGOD7** [DeathGOD7/unexpectedspawn-paper#6]
Unexpected Spawn is really unexpected....haha (sorry for my sarcasm...lets get to the real bug)

I tried your spigot page release : It got all things random spawn when dying, random spawning when first join and always random spawn or blacklist. But it got nether bug that it doesn't count the respawn anchor... and there are more bugs 😄 

Tested in : 1.16.5 Paper

Here are the things it does:
- If you have bed respawn in overworld, and you die in nether -> you tp back to bed (its good)
- If you don't have bed respawn in overworld , and you die in nether -> you get tp to random location in nether but above bedrock
- same applies to if you have respawn anchor(which overrides your bed respawn) -> you get tp to random location in nether but above bedrock

**Additional : I tried your reply in this [issue](https://github.com/Shivelight/unexpectedspawn-paper/issues/5#issuecomment-835836236). It teleports you to lava or any block which is located at 0 y 0 position of nether.**

And also there are no option in config to disable random respawn or blacklist world or per world respawn/first random spawn/join random spawn toggle.

Here are tips if you want to fix/implement those features.

For better blacklist world or per world settings, please add these :
```
# UnexpectedSpawn
# https://github.com/Shivelight/unexpectedspawn-paper
# Please change "survival" to the name of your world

global :
      x-max: 399
      x-min: -399
      z-max: 399
      z-min: -399
      bed-respawn-enabled: true
	  random-respawn : false
      first-join-only: false
      always-on-join: false
      spawn-block-blacklist:
        - LAVA
        - WATER
        - CACTUS	

# If no worlds are specified, it will use global/default variables. Default Config (worlds: [])
# If you have any world added below, it will override the global settings.
# If it got missing parameters that is in global settings like "spawn-block-blacklist"
# but not in worlds world parameters then it will use global parameters.
worlds : 
	survival :
		x-max: 399
		x-min: -399
		z-max: 399
		z-min: -399
		bed-respawn-enabled: true
		random-respawn : true
		first-join-only: true
		always-on-join: false
		
# If you have any worlds here , then it will be excluded from having random spawn
# Even if you have set custom settings in above settings and you add that world to 
# blacklist, it will be excluded. Default :[]		
blacklisted-worlds:		
	- bedwars
	- creative
```

About respawn anchor :
- If you have bed respawn point set in overworld and you die in nether -> you tp to bed (its already there)
- If you have respawn anchor (which overrides bed spawn point) and you die in nether -> you tp to respawn anchor  
- If you don't have bed respawn point set + also don't have respawn anchor and you die in nether -> you tp to overworld but in random place specified in global or its own world settings. 

I know its long but I hope I wrote it clearly.
Thank You
- Death GOD 7

**Comment by Shivelight** [comment]
Thank you, these are valuable input.

Multi-world setup is outside the scope of this plugin. I create this plugin for my old server (and learning) and strictly for vanilla/anarchy setup (1 world set). But I'll consider :+1: 

> If you don't have bed respawn in overworld , and you die in nether -> you get tp to random location in nether but above bedrock

I don't have this issue on my local test server. Can you share your plugin list? I think there is a plugin conflict.

> same applies to if you have respawn anchor(which overrides your bed respawn) -> you get tp to random location in nether but above bedrock

This is weird, the plugin simply ignores the anchor respawn event and let the server (or other plugins) handle it. (92bf6af0c2f8c7144ba6b182ab700ec62787f43c)

By the way, I now understand that respawn anchor and bed are mutually exclusive. I really thought they are alternating between overworld and nether (this is what will happen if you develop a plugin while **not** playing the game).

EDIT: Tested using the version I uploaded on the other issue.

**Comment by RexSE7EN** [comment]
```
If you don't have bed respawn in overworld , and you die in nether -> you get tp to random location in nether but above bedrock

 >  I don't have this issue on my local test server. Can you share your plugin list? I think there is a plugin conflict.
```
Plugins I used: Essentials, EssentialsChat (I have tried with both essentials removed), Multiverse-Core, Multiverse-Portals, StayPut, Towny, TownyChat, UnexpectedSpawn(and also used your dev build from another issue), Vault, WorldEdit, WorldGuard.

If  I use the spigot release 0.2.0, I tp to bedrock. Maybe cause of essential X safe tp or it does random tp in nether. But It happened in fresh install also.
And the development one in another issue caused me to tp to 0,y,0. 

I hope it's fixed in 0.2.1.
Thank you for making this plugin. Hope you are safe in this pandemic and have a nice day.
Love from Nepal.

**Comment by RexSE7EN** [comment]
Hi again, in v 0.2.1 it still teleports me to this coords in nether when I die (when I had respawn anchor):
![image (1)](https://user-images.githubusercontent.com/36422471/123647093-84f72880-d847-11eb-829c-0309d86fcf9d.png)


And also please add blacklist world in config as I don't want to do random tp in pvp arena or hub.

**### UPDATE**
I tried with a fresh install with this plugin only and the respawn anchor worked. It conflicted with ~~Essentials X~~ Multiverse old ver below 4.2.2 (please use ver above 4.3.0 to fix it). Thank you

**Comment by Shivelight** [comment]
Sorry for the late response.

> Thank you for making this plugin. Hope you are safe in this pandemic and have a nice day.
Love from Nepal.

Hey, I hope you are safe too!

> Maybe it conflicted with Essentials X.

Or maybe it is conflicting with Multiverse instead of Essential X? I read both plugins source code earlier and I remember that only Multiverse handles respawn event. CMIIW.

**Comment by RexSE7EN** [comment]
> Sorry for the late response.
> 
> > Thank you for making this plugin. Hope you are safe in this pandemic and have a nice day.
> > Love from Nepal.
> 
> Hey, I hope you are safe too!
> 
> > Maybe it conflicted with Essentials X.
> 
> Or maybe it is conflicting with Multiverse instead of Essential X? I read both plugins source code earlier and I remember that only Multiverse handles respawn event. CMIIW.

**Update :** 
Indeed it was multiverse. 
You will respawn in the spawn point of Nether world if you are using the old version of Multiverse 4.2.2 and below and running a 1.16+ server.
To fix that just use the latest build or v 4.3.0 or higher where multiverse fixed that issue.
